models/credmark/protocols/lending/aave/aave_v2.py

```diff
@@ -1,83 +1,120 @@
     aaveDebtInfos: List[AaveDebtInfo]
     _iterator: str = 'aaveDebtInfos'
 
 
+# For different markets
+AAVE_LENDING_POOL_ADDRESS_PROVIDER_REGISTRY = '0x52D306e36E3B6B02c153d0266ff0f85d18BCD413'
+
+# For main market
+AAVE_LENDING_POOL_ADDRESS_PROVIDER = '0xb53c1a33016b2dc2ff3653530bff1848a515c8c5'
+
 AAVE_LENDING_POOL_V2 = '0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9'
 
 
-@Model.describe(slug="aave.overall-liabilities-portfolio",
+@Model.describe(slug="aave-v2.get-lending-pool",
+                version="1.0",
+                display_name="Aave V2 - Get lending pool for main market",
+                description="Aave V2 - Get lending pool for main market",
+                input=EmptyInput,
+                output=Contract)
+class AaveV2GetLendingPool(Model):
+    def run(self, input: EmptyInput) -> Contract:
+        address_provider = Contract(address=Address(AAVE_LENDING_POOL_ADDRESS_PROVIDER).checksum)
+        lending_pool_address = address_provider.functions.getLendingPool().call()
+        lending_pool_contract = Contract(address=lending_pool_address)
+        return lending_pool_contract
+
+
+@Model.describe(slug="aave-v2.get-price-oracle",
+                version="1.0",
+                display_name="Aave V2 - Get price oracle for main market",
+                description="Aave V2 - Get price oracle for main market",
+                input=EmptyInput,
+                output=Contract)
+class AaveV2GetPriceOracle(Model):
+    def run(self, input: EmptyInput) -> Contract:
+        address_provider = Contract(address=Address(AAVE_LENDING_POOL_ADDRESS_PROVIDER).checksum)
+        price_oracle_address = address_provider.functions.getPriceOracle().call()
+        price_oracle_contract = Contract(address=price_oracle_address)
+        return price_oracle_contract
+
+
+@Model.describe(slug="aave-v2.overall-liabilities-portfolio",
                 version="1.0",
                 display_name="Aave V2 Lending Pool overall liabilities",
                 description="Aave V2 liabilities for the main lending pool",
                 output=Portfolio)
 class AaveV2GetLiability(Model):
 
     def run(self, input) -> Portfolio:
-        contract = Contract(
-            address=Address(AAVE_LENDING_POOL_V2).checksum,
-            abi=AAVE_V2_TOKEN_CONTRACT_ABI
-        )
+        aave_lending_pool = self.context.run_model('aave-v2.get-lending-pool',
+                                                   input=EmptyInput(),
+                                                   return_type=Contract)
 
-        aave_assets = contract.functions.getReservesList().call()
+        aave_assets = aave_lending_pool.functions.getReservesList().call()
 
         positions = []
         for asset in aave_assets:
             # self.logger.info(f'getting info for {asset=}')
-            pos = self.context.run_model(slug='aave.token-liability',
+            pos = self.context.run_model(slug='aave-v2.token-liability',
```

Comment:
do we prefer the 

`self.context.run_model()` or `self.context.models.aave_v2.token_liability` syntax?
Either works, I think it's a preference thing.

Reply:
However, there is one difference between the two: run_model() can specify the return type to be in DTO, instead of a dictionary.